sdk/core/azure_core/README.md

```diff
@@ -1,20 +1,283 @@
 
-License: MIT
+Typically, you will not need to install `azure_core`;
+it will be installed for you when you install one of the client libraries using it.
+In case you want to install it explicitly (to implement your own client library, for example),
+you can find the crates.io package [here][Package (crates.io)].
+
+## Key concepts
+
+The main shared concepts of `azure_core` (and so Azure SDK libraries using `azure_core`) include:
+
+- Configuring service clients, e.g. configuring retries, logging (`ClientOptions`).
+- Accessing HTTP response details (`Response`, `Response<T>`).
+- Calling long-running operations (`Operation<T>`).
+- Paging and asynchronous streams (`AsyncPageable<T>`).
+- Exceptions for reporting errors from service requests in a consistent fashion. (`RequestFailedException`).
+- Customizing requests (`RequestContext`).
+- Abstractions for representing Azure SDK credentials. (`TokenCredentials`).
+
+Below, you will find sections explaining these shared concepts in more detail.
+
+### Thread safety
+
+We guarantee that all client instance methods are thread-safe and independent of each other ([guideline](https://azure.github.io/azure-sdk/rust_introduction.html#rust-service-methods-thread-safety)). This ensures that the recommendation of reusing client instances is always safe, even across threads.
+
+### Additional concepts
+<!-- CLIENT COMMON BAR -->
+[Client options](#configuring-service-clients-using-clientoptions) |
+[Accessing the response](#accessing-http-response-details-using-responset) |
+[Long-running operations](#consuming-long-running-operations-using-operationt) |
+<!-- [Handling failures](#reporting-errors-requestfailedexception) |
+[Diagnostics](https://github.com/Azure/azure-sdk-for-rust/blob/main/sdk/core/azure_core/samples/Diagnostics.md) |
+[Mocking](https://learn.microsoft.com/rust/azure/sdk/unit-testing-mocking) |
+[Client lifetime](https://devblogs.microsoft.com/azure-sdk/lifetime-management-and-thread-safety-guarantees-of-azure-sdk-rust-clients/) -->
+<!-- CLIENT COMMON BAR -->
+
+## Examples
+
+**NOTE:** Samples in this file apply only to packages that follow [Azure SDK Design Guidelines](https://azure.github.io/azure-sdk/rust_introduction.html). Names of such packages usually start with `azure_`.
+
+### Configuring Service Clients Using `ClientOptions`
+
+Azure SDK client libraries typically expose one or more _service client_ types that
+are the main starting points for calling corresponding Azure services.
+You can easily find these client types as their names end with the word _Client_.
+For example, `SecretClient` can be used to call the Key Vault service and interact with secrets,
+and `KeyClient` can be used to access Key Vault service cryptographic keys.
+
+These client types can be instantiated by calling a simple `new` or `builder` function that takes various configuration options.These options are passed as a parameter that extends `ClientOptions` class exposed by `azure_core`.
```

Comment:
I wouldn't mention builders. Only a few will use it, but it wouldn't be common enough to mention in `azure_core` I think.

Reply:
Removed mention builders